alacritty/src/renderer/mod.rs

```diff
@@ -392,29 +392,29 @@ impl GlyphCache {
 #[repr(C)]
 struct InstanceData {
     // Coords.
-    col: f32,
-    row: f32,
+    col: u16,
+    row: u16,
     // Glyph offset.
-    left: f32,
-    top: f32,
+    left: i16,
+    top: i16,
     // Glyph scale.
-    width: f32,
-    height: f32,
+    width: i16,
+    height: i16,
     // uv offset.
     uv_left: f32,
     uv_bot: f32,
     // uv scale.
     uv_width: f32,
     uv_height: f32,
     // Color.
-    r: f32,
-    g: f32,
-    b: f32,
+    r: u8,
+    g: u8,
+    b: u8,
```

Comment:
Based on some quick testing on my machine, these seem to be the biggest problem.

Reply:
Woo, nice to see this land. Was this the main issue causing a performance degradation?

Reply:
Yeah, I believe so. It was just some really weird edge case with the types macOS accepts.

I just tested a bunch of different variations and moving it to separate u8 fields just worked best.

Reply:
I wish I could say I understood the issue and to tell you *why* macOS doesn't like these specific types, but really I just brute-forced it.

Reply:
macOS :roll_eyes: 

Impressive work on finding the issue